src/sentry/api/endpoints/integrations/sentry_apps/rotate_secret.py

```diff
@@ -0,0 +1,62 @@
+from django.http import Http404
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import control_silo_endpoint
+from sentry.api.bases.sentryapps import SentryAppBaseEndpoint
+from sentry.api.permissions import SentryPermission
+from sentry.api.serializers import serialize
+from sentry.models.apiapplication import generate_token
+from sentry.models.integrations.sentry_app import SentryApp
+from sentry.services.hybrid_cloud.organization import organization_service
+from sentry.services.hybrid_cloud.user.service import user_service
+
+
+class SentryAppRotateSecretPermission(SentryPermission):
+    scope_map = {
+        "POST": ["org:write", "org:admin"],
+    }
+
+    def has_object_permission(self, request: Request, view: object, sentry_app: SentryApp):
+        # organization that owns an integration
+        org_context = organization_service.get_organization_by_id(id=sentry_app.owner_id)
+        if org_context is None:
+            raise Http404
+
+        organization = org_context.organization
+        if organization is None:
+            raise Http404
+
+        # if user is not a member of an organization owning an integration,
+        # return 404 to avoid leaking integration slug
+        organizations = (
+            user_service.get_organizations(user_id=request.user.id)
+            if request.user.id is not None
+            else ()
+        )
+        if not any(organization.id == org.id for org in organizations):
+            raise Http404
+
+        # permission check inside an organization
+        self.determine_access(request, organization)
+        allowed_scopes = set(self.scope_map.get(request.method or "", []))
+        return any(request.access.has_scope(s) for s in allowed_scopes)
+
+
+@control_silo_endpoint
+class SentryAppRotateSecretEndpoint(SentryAppBaseEndpoint):
+    publish_status = {
+        "POST": ApiPublishStatus.PRIVATE,
+    }
+    owner = ApiOwner.ENTERPRISE
+    permission_classes = (SentryAppRotateSecretPermission,)
+
+    def post(self, request: Request, sentry_app: SentryApp) -> Response:
+        if sentry_app.application is None:
+            return Response(status=404)
+
+        new_token = generate_token()
+        sentry_app.application.update(client_secret=new_token)
```

Comment:
should we add a try/catch here and respond accordingly? And also log the error? Because this is a DB transaction, there could be errors